RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitConvertGPU2CPUFormat.cc

```diff
@@ -19,73 +19,82 @@ class EcalUncalibRecHitConvertGPU2CPUFormat : public edm::stream::EDProducer<> {
   void produce(edm::Event&, edm::EventSetup const&) override;
 
 private:
+  bool isPhase2_;
   const edm::EDGetTokenT<InputProduct> recHitsGPUEB_;
-  const edm::EDGetTokenT<InputProduct> recHitsGPUEE_;
+  edm::EDGetTokenT<InputProduct> recHitsGPUEE_;
 
-  const std::string recHitsLabelCPUEB_, recHitsLabelCPUEE_;
+  const std::string recHitsLabelCPUEB_;
+  std::string recHitsLabelCPUEE_;
 };
 
 void EcalUncalibRecHitConvertGPU2CPUFormat::fillDescriptions(edm::ConfigurationDescriptions& confDesc) {
   edm::ParameterSetDescription desc;
 
   desc.add<edm::InputTag>("recHitsLabelGPUEB", edm::InputTag("ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEB"));
-  desc.add<edm::InputTag>("recHitsLabelGPUEE", edm::InputTag("ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEE"));
 
   desc.add<std::string>("recHitsLabelCPUEB", "EcalUncalibRecHitsEB");
+
+  desc.add<bool>("isPhase2", false);
+
+  desc.add<edm::InputTag>("recHitsLabelGPUEE", edm::InputTag("ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEE"));
   desc.add<std::string>("recHitsLabelCPUEE", "EcalUncalibRecHitsEE");
 
   confDesc.add("ecalUncalibRecHitConvertGPU2CPUFormat", desc);
 }
 
 EcalUncalibRecHitConvertGPU2CPUFormat::EcalUncalibRecHitConvertGPU2CPUFormat(const edm::ParameterSet& ps)
-    : recHitsGPUEB_{consumes<InputProduct>(ps.getParameter<edm::InputTag>("recHitsLabelGPUEB"))},
-      recHitsGPUEE_{consumes<InputProduct>(ps.getParameter<edm::InputTag>("recHitsLabelGPUEE"))},
```

Comment:
Can you keep this and the other parameters as `const` and initialise them in the initialiser list ?
For example
```c++
       recHitsGPUEE_{isPhase2_ ? consumes<InputProduct>(ps.getParameter<edm::InputTag>("recHitsLabelGPUEE")) : edm::EDGetTokenT<InputProduct>{}},
```